src/pages/gd/Swap/SwapConfirmModal/index.tsx

```diff
@@ -65,28 +65,67 @@ function SwapConfirmModal({
     const web3 = useWeb3()
     const [status, setStatus] = useState<'PREVIEW' | 'CONFIRM' | 'SENT'>('SENT')
     const [hash, setHash] = useState('')
+    const [sender, setSender] = useState('')
+
+    const senderRef = useRef(sender)
+
+    useEffect(() => {
+      senderRef.current = sender
+    }, [sender])
 
     const handleSwap = async () => {
         setStatus('CONFIRM')
+
+        const prepareTx = (from: string) => {
+          setSender(from)
+        }
+        
         const onSent = (hash: string) => {
             setStatus('SENT')
             setHash(hash)
-            if (onConfirm) onConfirm()
-        }
-        try {
-            let transactionDetails = buying ? await buy(web3!, meta!, onSent) : await sell(web3!, meta!, onSent)
+            const inputSig = meta?.inputAmount.toSignificant(5)
+            const minimumOutputSig = meta?.minimumOutputAmount.toSignificant(5)
+
+            const tradeInfo = {
+              input: {
+                decimals: meta?.inputAmount.currency.decimals,
+                symbol: meta?.inputAmount.currency.symbol
+              },
+              output: {
+                decimals: meta?.outputAmount.currency.decimals,
+                symbol: meta?.outputAmount.currency.symbol
+              }
+            }
+            const summary = 'Swapping ' + inputSig + ' ' 
+                            + meta?.inputAmount.currency.symbol + 
+                            ' to a minimum of ' + minimumOutputSig + ' ' +
+                            meta?.outputAmount.currency.symbol
             globalDispatch(
-                addTransaction({
-                    chainId: chainId!,
-                    hash: transactionDetails.transactionHash,
-                    from: transactionDetails.from
-                })
+              addTransaction({
+                chainId: chainId!,
+                hash: hash,
+                from: senderRef.current,
+                summary: summary,
+                tradeInfo: tradeInfo,
+              })
             )
+            if (onConfirm) onConfirm()
+        }
+        try { 
+            buying ? await buy(web3!, meta!, prepareTx, onSent) : await sell(web3!, meta!, prepareTx, onSent)
+            // let transactionDetails = buying ? await buy(web3!, meta!, prepareTx, onSent) : await sell(web3!, meta!, prepareTx, onSent)
+            // globalDispatch(
+            //     addTransaction({
+            //         chainId: chainId!,
+            //         hash: transactionDetails.transactionHash,
+            //         from: transactionDetails.from
+            //     })
+            // )
```

Comment:
why was this replaced? it looks more correct to me... to add the transaction after it is sent to the blockchain